fe/fe-core/src/main/java/com/starrocks/sql/optimizer/task/TopDownIterativeTask.java

```diff
@@ -0,0 +1,81 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021 StarRocks Limited.
+package com.starrocks.sql.optimizer.task;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.starrocks.sql.optimizer.Group;
+import com.starrocks.sql.optimizer.GroupExpression;
+import com.starrocks.sql.optimizer.OptExpression;
+import com.starrocks.sql.optimizer.operator.pattern.Pattern;
+import com.starrocks.sql.optimizer.rule.Binder;
+import com.starrocks.sql.optimizer.rule.Rule;
+import com.starrocks.sql.optimizer.rule.RuleSetType;
+import com.starrocks.sql.optimizer.rule.RuleType;
+
+import java.util.List;
+
+/**
+ * TopDownIterativeTask performs a top-down rewrite logic operator pass.
+ * <p>
+ * This class modify from CMU noisepage project TopDownRewrite class
+ */
+public class TopDownIterativeTask extends OptimizerTask {
```

Comment:
This class could merge to `TopDownRewriteTask`

Reply:
done